zipline/_protocol.pyx

```diff
@@ -21,6 +21,28 @@ from cpython cimport bool
 
 from zipline.assets import Asset
 
+
+class assert_keywords(object):
+    """
+    Asserts that the keywords passed into the wrapped function are included
+    in those passed into this decorator. If not, raise a TypeError with a
+    meaningful message, unlike the one cython returns by default.
+    """
+
+    def __init__(self, *args):
+        self.names = args
```

Comment:
If this is only every checking membership, could use a `set` here too. But the number of elements is so small, that the tradeoff between checking every element is probably marginal.